### PR TITLE
Validate that function names are valid java identifiers

### PR DESCRIPTION
**Pull request by dguy**
### Description 
Validate that function names are:
1. not a java reserved word
2. not a KSQL reserved word
3. not null or empty
4. is a valid java identifier

### Testing done 
Added unit test for new flass

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-engine/src/main/java/io/confluent/ksql/function/FunctionNameValidator.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  private static final Set<String> KSQL_RESERVED_WORDS
+      = ImmutableSet.<String>builder()
+      .add("analyze").add("as").add("catalog").add("create").add("beginning").add("by")
+      .add("delete").add("describe").add("drop").add("exists").add("explain").add("export")
+      .add("extended").add("insert").add("into").add("join").add("from").add("list").add("load")
+      .add("limit").add("not").add("partition").add("print").add("properties").add("queries")
+      .add("registered").add("run").add("script").add("select").add("set").add("show")
+      .add("stream").add("streams").add("table").add("tables").add("terminate")
+      .add("topics").add("unset").add("when").add("where").add("window").add("with")
+      .build();
```
Comment (hjafarpour) [inline ksql-engine/src/main/java/io/confluent/ksql/function/FunctionNameValidator.java]: Seems that some of the reserved words such as `GROUP`, `LEFT' and `CAST` are missing. You can find all of them in the grammar or even a better source would be the generated token file.

Comment (dguy) [reply ksql-engine/src/main/java/io/confluent/ksql/function/FunctionNameValidator.java]: Yeah - that is why i left the comment. Where does the token file get generated to?

Comment (dguy) [reply ksql-engine/src/main/java/io/confluent/ksql/function/FunctionNameValidator.java]: found it

Comment (blueedgenick) [reply ksql-engine/src/main/java/io/confluent/ksql/function/FunctionNameValidator.java]: feels very brittle ? normally the generated SqlBaseParser.java class will contain a map of all the grammar symbols - is that something we could use dynamically here ?

File: ksql-engine/src/main/java/io/confluent/ksql/function/FunctionNameValidator.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      .add("throws").add("transient").add("try").add("short").add("static").add("strictfp")
+      .add("super").add("switch").add("synchronized").add("void").add("volatile").add("while")
+      .build();
+
+  // This has been generated from SqlBase.tokens
+  private static final Set<String> KSQL_RESERVED_WORDS
+      = ImmutableSet.<String>builder()
+      .add("add").add("advance").add("all").add("alter").add("analyze").add("and").add("any")
+      .add("approximate").add("array").add("as").add("asc").add("asterisk").add("at")
+      .add("backquoted_identifier").add("beginning").add("bernoulli").add("between")
```
Comment (apurvam) [inline ksql-engine/src/main/java/io/confluent/ksql/function/FunctionNameValidator.java]: 'bernoulli', 'cross,  'nfd', 'nkfc', etc. etc. this makes me think that the list was autogenerated from the lexer tokens. There is way to get this automatically from the `SqlBaseParser.vocabulary`public static field. That would be a more robust way to do this IMO.

File: ksql-engine/src/main/java/io/confluent/ksql/function/FunctionNameValidator.java (hunk 1 above)
Comment (apurvam) [reply ksql-engine/src/main/java/io/confluent/ksql/function/FunctionNameValidator.java]: ah. I just saw this thread. Yes, the lexical tokens are available in the code through the `SqlBaseParser.vocabulary` field. We should just use that.

File: ksql-engine/src/main/java/io/confluent/ksql/function/FunctionNameValidator.java (hunk 2 above)
Comment (dguy) [reply ksql-engine/src/main/java/io/confluent/ksql/function/FunctionNameValidator.java]: Cool - thanks. Should've looked there

**Pull request merged**